File: app/models/shipping_way.rb
class ShippingWay < ActiveHash::Base
  self.data = [
    {id: 1, name: '未定', group: "1"},
    {id: 2, name: 'ゆうパック', group: "1"},
    {id: 3, name: 'クロネコヤマト', group: "1"},
    {id: 4, name: 'ネコポス', group: "1"},
    {id: 5, name: '飛脚メール便', group: "1"},
    {id: 6, name: '宅急便コンパクト', group: "1"},
    {id: 7, name: 'レターパックライト', group: "1"},
    {id: 8, name: 'レターパックプラス', group: "1"},
    {id: 9, name: 'ゆうパケット', group: "1"},
    {id: 10, name: 'スマートレター', group: "1"},
    {id: 11, name: 'クリックポスト', group: "1"},
    {id: 12, name: 'ゆうメール', group: "1"},
    {id: 13, name: '普通郵便（定形、定形外）', group: "1"},

    {id: 14, name: '未定', group: "2"},
    {id: 15, name: 'ゆうパック', group: "2"},
    {id: 16, name: 'ゆうメール', group: "2"},
    {id: 17, name: 'クロネコヤマト', group: "2"},
  ]
end

File: app/assets/javascripts/shipping-way.js
function appendOption(shipping_way) {
  var html = `<option value="${shipping_way.id}">${shipping_way.name}</option>`;
  return html;
}
function selectShippingWayHtml(insertHtml){
  var html = `<div class='product-select-wrapper' id= 'shipping-way_wrapper'>
                <h3>
                  配送方法
                  <span class='form-required'>
                  必須
                  </span>
                </h3>
                <div class='items-data--shipping-way' >
                  <select class="select-box" id="shipping-way" name="item[shipping_way_id]">
                    ${insertHtml}
                  </select>
                </div>
              </div>`;
  return html; 
}


$(function() {
  $(document).on('change',"#shipping-fee_select", function() {
    var selected_fee = $(this).val();
    if(selected_fee != ""){
      $.ajax({
        url: '/items/get_shipping_way',
        type: "GET",
        data: { selected_fee: selected_fee},
        dataType: 'json',
      })
      .done(function(shipping_ways){
        $('#shipping-way_wrapper').remove();
        var insertHtml = '';
        shipping_ways.forEach(function(shipping_way){
          insertHtml += appendOption(shipping_way);
        });
        $('.delivery--way').append(selectShippingWayHtml(insertHtml));
      })
      .fail(function(){
        alert('配送料の負担の取得エラー');
      })
    } else {
      $('#shipping-way_wrapper').remove();
    };
  });
});
// render用
// $(window).load(function () {
//   var selected_fee = $("#shipping-fee_select").val();
//   if(selected_fee != ""){
//     $.ajax({
//       url: '/items/get_shipping_way',
//       type: "GET",
//       data: { selected_fee: selected_fee},
//       dataType: 'json',
//     })
//     .done(function(shipping_ways){
//       var insertHtml = '';
//       console.log(shipping_ways);
//       shipping_ways.forEach(function(shipping_way){
//        insertHtml += appendOption(shipping_way);
//       });
//       $('.delivery--way').append(selectShippingWayHtml(insertHtml));
//     })
//     .fail(function(){
//       alert('配送料の負担の取得エラー');
//     })
//   };
// });